perian/perian-core/src/utils/plugin.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import type { Plugin } from '../../types/types.js'
+import { addNetwork } from '../commons/network.js'
+
+export const addPlugin = (moduleName: Plugin) => {
+  if (Array.isArray(moduleName)) {
+    moduleName.forEach(addPlugin)
+  } else {
+    try {
+      if (!moduleName.startsWith('@perian/')) {
+        throw new Error('Unknown plugin name')
+      }
+      const plugin = moduleName.replace('@perian/', '').split('-')
+      if (plugin[0] === 'network') {
+        // $FlowFixMe
+        addNetwork(plugin[1], require(moduleName))
+      }
+    } catch (e) {
+      console.log(e)
```

Comment:
This catch & log pattern only makes sense if you want to keep going after failure. Is that what's wanted? Is it really OK for coins to be missing?

Reply:
removed